cognite/client/data_classes/data_modeling/_validation.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+RESERVED_EXTERNAL_IDS = frozenset(
+    {
+        "Query",
+        "Mutation",
+        "Subscription",
+        "String",
+        "Int32",
+        "Int64",
+        "Int",
+        "Float32",
+        "Float64",
+        "Float",
+        "Timestamp",
+        "JSONObject",
+        "Date",
+        "Numeric",
+        "Boolean",
+        "PageInfo",
+        "File",
+        "Sequence",
+        "TimeSerie",
+    }
+)
+RESERVED_SPACE_IDS = frozenset({"space", "cdf", "dms", "pg3", "shared", "system", "node", "edge"})
+
+RESERVED_PROPERTIES = frozenset(
+    {
+        "space",
+        "externalId",
+        "createdTime",
+        "lastUpdatedTime",
+        "deletedTime",
+        "edge_id",
+        "node_id",
+        "project_id",
+        "property_group",
+        "seq",
+        "tg_table_name",
+        "extensions",
+    }
+)
+
+
+def validate_data_modeling_identifier(space: str | None, external_id: str | None = None) -> None:
```

Comment:
Consider moving to `_validation.py`

Reply:
@erlendvollset Asked me to move it here.